astropy/utils/introspection.py

```diff
@@ -154,6 +155,12 @@ def minversion(module, version, inclusive=True, version_path='__version__'):
         from pkg_resources import parse_version
     except ImportError:
         from distutils.version import LooseVersion as parse_version
+        # LooseVersion raises a TypeError when strings like dev, rc1 are part
+        # of the version number. Match the dotted numbers only.
+        expr = '^([1-9]\\d*!)?(0|[1-9]\\d*)(\\.(0|[1-9]\\d*))*'
```

Comment:
The regex is lifted straight from PEP440, which defines valid version numbers. I had the same question as you when I read it but decided to go with the "official" version in case I had missed something. My brain's regex parser is notoriously unreliable.